core/src/main/java/org/zenoss/zep/dao/impl/EventSummaryDaoImpl.java

```diff
@@ -228,7 +228,14 @@ public String create(Event event, final EventPreCreateContext context) throws Ze
         final byte[] fingerprintHash;
         final String uuid;
         if (ZepConstants.CLOSED_STATUSES.contains(event.getStatus())) {
-            fingerprintHash = DaoUtils.sha1(fingerprint + '|' + System.currentTimeMillis());
+            long ts;
+            ts = System.currentTimeMillis();
+            // Change fingerprintHash value for clear event to prevent
+            // undesirable DuplicateKeyException (ZEN-26244)
+            if (event.getSeverity() == EventSeverity.SEVERITY_CLEAR) {
```

Comment:
When the clear event and the event it clears happen to close in the same millisecond, the fingerprint hashes will be the same and the latter event won't close because a DuplicateKeyException is raised (fingerprint hash is a unique key).